test/cast/any_of_test.exs

```diff
@@ -84,15 +85,48 @@ defmodule OpenApiSpex.CastAnyOfTest do
       assert {:ok, %{breed: "Corgi", age: 3, food: "meat"}} = cast(value: value, schema: schema)
     end
 
+    test "should proper return the errors if no match" do
+      schema_1 = %Schema{
+        type: :object,
+        additionalProperties: false,
+        properties: %{
+          breed: %Schema{type: :string}
+        }
+      }
+
+      schema_2 = %Schema{
+        type: :object,
+        additionalProperties: false,
+        properties: %{
+          food: %Schema{type: :string, minLength: 5}
+        }
+      }
+
+      schema = %Schema{anyOf: [schema_1, schema_2], title: "MyCoolSchema", type: :object}
+      value = %{"food" => "meat"}
+
+      assert {:error, [error_any_of, error_unexpected_field, error_food]} =
+               cast(value: value, schema: schema)
+
+      assert Error.message(error_any_of) ==
+               "Failed to cast value using any of: Schema(type: :object), Schema(type: :object)"
+
+      assert Error.message(error_unexpected_field) ==
```

Comment:
💡 Our errors should point to the schema.

This error for example, can be confusing as we'd have to guess it's from `schema_a`.

```elixir
   %OpenApiSpex.Cast.Error{
     format: nil,
     length: 0,
     meta: %{},
     name: "food",
     path: ["food"],
     reason: :unexpected_field,
     type: nil,
     value: %{"food" => "meat"}
   }
```

Reply:
Maybe this can be addressed in a later change?

Reply:
Yes, there doesn't seem to be an appropriate error field for this unless if we add this info in meta. I'll create a separate issue.